nmagent/responses.go

```diff
@@ -37,8 +43,74 @@ type NCVersionList struct {
 	Containers []NCVersion `json:"networkContainers"`
 }
 
+// HomeAZFix is an indication that a particular bugfix has been applied to some
+// HomeAZ.
+type HomeAZFix int
+
+func (h HomeAZFix) String() string {
+	switch h {
+	case HomeAZFixInvalid:
+		return "HomeAZFixInvalid"
+	case HomeAZFixIPv6:
+		return "HomeAZFixIPv6"
+	default:
+		return "Unknown HomeAZ Fix"
+	}
+}
+
+const (
+	HomeAZFixInvalid HomeAZFix = iota
+	HomeAZFixIPv6
+)
+
 type AzResponse struct {
-	HomeAz uint `json:"homeAz"`
+	HomeAz       uint
+	AppliedFixes []HomeAZFix
```

Comment:
Hard disagree. From everything I've seen, the meaning of this uint is to describe fixes that have been applied that would be of interest to a client for whatever reason. It may be *easier* to have a uint here, but it's also a horrible user experience that spreads esoteric knowledge about magic numbers throughout multiple codebases. Yes, today `apiVersion` can be `0` or `2`, but it is better engineering to assume that there will be future values for this to convey other fixes that have been applied. This is a slice because, in my experience in this industry, sooner or later there will be more than one fix of interest to clients. This is here to accommodate that eventuality.

This design is more robust, because however this contract changes in the future will require changes in **exactly one place** without breaking clients.

Reply:
Okay. Let's keep it this way then.  It is more future proof, whether we need it right now or not.